### DISCUSSION TITLE
What is the different between `0x800` and `-2048`

### DISCUSSION THREAD

**Discussion by ranvd** [xpack-dev-tools/riscv-none-elf-gcc-xpack#4] [Q&A] [answered]
I am using `riscv-none-elf-gcc -c` command to compile RISC-V instruction `ori x7, x8, 0x800` and it show an error `Error: illegal operands ori x7,x8,0x800'`. But, if I change the immediate into decimal format `ori x7, x8, -2048`. It will compile correctly. What is the different between these two format.

I am sorry if this is a stupid question.

**Comment by ilg-ul** [comment, marked as answer]
I regret but I don't know why only signed values are accepted, probably this is how the instructions are defined, with 12-bit signed immediate values and 0x800 is interpreted as positive 2048, which is outside the range.

For details related to the assembler syntax, please ask on the RISC-V support forums.

**Reply by ranvd** [reply]
Oh, I should have to notice that. I thought there is no signed values for hex format. Thank you.
